Content.Client/Radar/RadarConsoleWindow.xaml.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using Content.Client.Computer;
+using Content.Shared.Radar;
+using JetBrains.Annotations;
+using Robust.Client.AutoGenerated;
+using Robust.Client.GameObjects;
+using Robust.Client.Graphics;
+using Robust.Client.UserInterface;
+using Robust.Client.UserInterface.CustomControls;
+using Robust.Client.UserInterface.XAML;
+using Robust.Shared.Maths;
+
+namespace Content.Client.Radar;
+
+[GenerateTypedNameReferences]
+public partial class RadarConsoleWindow : DefaultWindow, IComputerWindow<RadarConsoleBoundInterfaceState>
+{
+    public RadarConsoleWindow()
+    {
+        RobustXamlLoader.Load(this);
+    }
+
+    public void SetupComputerWindow(ComputerBoundUserInterfaceBase cb)
+    {
+
+    }
+
+    public void UpdateState(RadarConsoleBoundInterfaceState scc)
+    {
+        Radar.UpdateState(scc);
+    }
+}
+
+
+public sealed class RadarControl : Control
+{
+    private const int RadarArea = 512;
+
+    private const int RadarCircleRadius = (RadarArea - 8) / 2;
+
+    private RadarConsoleBoundInterfaceState _lastState = new RadarConsoleBoundInterfaceState(new RadarObjectData[] { }, Vector2i.Zero);
+
+    private int SizeFull => (int) (RadarArea * UIScale);
+
+    public int RadiusCircle => (int) (RadarCircleRadius * UIScale);
+
+    public RadarControl()
+    {
+        MinSize = (SizeFull, SizeFull);
+    }
+
+    public void UpdateState(RadarConsoleBoundInterfaceState ls)
+    {
+        _lastState = ls;
+    }
+
+    protected override void Draw(DrawingHandleScreen handle)
+    {
+        var point = SizeFull / 2;
+        var fakeAA = new Color(0.08f, 0.08f, 0.08f);
+        var gridLines = new Color(0.08f, 0.08f, 0.08f);
+        var gridLinesRadial = 8;
+        var gridLinesEquatorial = 8;
+
+        handle.DrawCircle((point, point), RadiusCircle + 1, fakeAA);
+        handle.DrawCircle((point, point), RadiusCircle, Color.Black);
```

Comment:
Circles are somewhat expensive so if you can get away with using other shape types (e.g. triangles) it may be much more preferable for people with potatos.

Reply:
Wait meant to put the comment below on the radar bit but pretend I did.